dev-docs/RFCs/vNext/icon-layer-dynamic-image-sources.md

```diff
@@ -0,0 +1,52 @@
+# RFC: Icon Manager 
+
+* **Authors**: Xintong Xia 
+* **Date**: December 2018
+* **Status**: Draft 
+
+## Abstract
+
+This RFC proposes a way to allow IconLayer load image sources at runtime. 
+
+## Motivation 
+
+[IconLayer](/docs/layers/icon-layer.md) currently requires all icons pre-packed into a sprite image (`iconAtlas`) and a JSON descriptor (`iconMapping`). 
+In some use cases, it is not possible to know the icons that will be used. Instead, each icon needs to be fetched from
+a programmatically generated URL at runtime.
+ 
+## Proposal - Extending current `getIcon` API
+
+Change the `getIcon` API to allow user to return either a `string` or an `object`. 
```

Comment:
Clarify whether `iconAtlas` and `iconMapping` props are still required. This constitutes an API change too.